src/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -136,13 +143,54 @@ private void InitiateUnload()
 
         public bool IsCollectible { get; }
 
+        public string Name { get; }
+
+        public override string ToString() => "Name: " + Name + "Type: " + GetType().Name + "Id:" + _id;
```

Comment:
Does this need to be localized?

Reply:
Maybe we could format this in such a way that it doesn't need words...
`<name> [type] #id`
So for example
`PluginOne [PluginLoadContext] #4`

Since this is intended for diagnostic purposes, string like that should work just fine. It's also shorter which should work out better in debuggers.

Reply:
We should also handle null names in some nicer way. If I don't give it a name it might look like
`<type> #id` instead.
Or maybe even keep the existing behavior and simply print out the type name.

Reply:
I changed this to
- Quoted the name
- Type undecorated
- # prefix on id

Null name looks like an empty string.